### PR TITLE
Migrate ec2 vpc endpoint modules

### PR DESCRIPTION
**Pull request by jillr**
Depends-On: https://github.com/ansible/project-config/pull/862

##### SUMMARY
Moves ec2_vpc_endpoint, ec2_vpc_endpoint_info, and ec2_vpc_endpoint_service_info from community.aws to amazon.aws via git patch.

As discussed at https://github.com/ansible-collections/amazon.aws/pull/354#pullrequestreview-657545481, this PR initially demonstrates the result of the git patch process.  When that is approved, the git patch process will be repeated directly onto `main` to preserve git history.  This PR will then be rebased on the updated `main` with the runtime.yml, deprecation, and example documentation changes.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
ec2_vpc_endpoint
ec2_vpc_endpoint_info
ec2_vpc_endpoint_service_info

##### ADDITIONAL INFORMATION
Patch process:

```
git log --pretty=email --patch-with-stat --reverse --full-index --binary -- tests/integration/targets/ec2_vpc_endpoint* plugins/modules/ec2_vpc_endpoint* > ../../amazon/aws/patch

cd ../amazon/aws/

git am < patch
```

### CODE REVIEW

**Comment by jillr** [comment]
recheck

File: plugins/modules/ec2_vpc_endpoint_service_info.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+module: ec2_vpc_endpoint_service_info
+short_description: retrieves AWS VPC endpoint service details
+version_added: 1.5.0
+description:
+  - Gets details related to AWS VPC Endpoint Services.
+requirements: [ boto3 ]
```
Comment (tremble) [inline plugins/modules/ec2_vpc_endpoint_service_info.py:14]:
```suggestion
```

post #298 this isn't needed in amazon.aws (https://github.com/ansible-collections/community.aws/pull/569 is still pending)

File: plugins/modules/ec2_vpc_endpoint_info.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+
+DOCUMENTATION = r'''
+module: ec2_vpc_endpoint_info
+short_description: Retrieves AWS VPC endpoints details using AWS methods.
+version_added: 1.0.0
+description:
+  - Gets various details related to AWS VPC endpoints.
+  - This module was called C(ec2_vpc_endpoint_facts) before Ansible 2.9. The usage did not change.
+requirements: [ boto3 ]
```
Comment (tremble) [inline plugins/modules/ec2_vpc_endpoint_info.py:15]:
```suggestion
```

File: plugins/modules/ec2_vpc_endpoint.py [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+
+DOCUMENTATION = r'''
+module: ec2_vpc_endpoint
+short_description: Create and delete AWS VPC Endpoints.
+version_added: 1.0.0
+description:
+  - Creates AWS VPC endpoints.
+  - Deletes AWS VPC endpoints.
+  - This module supports check mode.
+requirements: [ boto3 ]
```
Comment (tremble) [inline plugins/modules/ec2_vpc_endpoint.py:17]:
```suggestion
```

**Review by tremble — CHANGES_REQUESTED** [review]
with #298 boto3 became the default requirement for modules, this isn't the case for community.aws yet

File: plugins/modules/ec2_vpc_endpoint_service_info.py (hunk 1 above)
Comment (jillr) [reply plugins/modules/ec2_vpc_endpoint_service_info.py:14]: Shoot, 569 is totally on me - thanks!  :)

**Comment by goneri** [comment]
I add a Depends-On: https://github.com/ansible/project-config/pull/862 to be sure we don't squash the commits during the PR landing :-).

**Comment by goneri** [comment]
@jillr, I've pushed https://github.com/ansible-collections/amazon.aws/pull/393/commits which is based on your branch.

The difference is that I add a footer in the commit message. It includes a URL pointing on the original commit.

I use the following script to build this up: https://github.com/goneri/promote_aws_modules

**Comment by jillr** [comment]
@goneri Should we close this and do all of the migrations with the automation? Or do you think it's worth landing this one since the PR is open?

File: plugins/modules/ec2_vpc_endpoint_service_info.py (hunk 1 above)
Comment (tremble) [reply plugins/modules/ec2_vpc_endpoint_service_info.py:14]: 569 has now been merged, not sure how you'd like to handle the updates here.

**Comment by goneri** [comment]
@jillr, we can actually close both and reopen a clean PR. I did the script as a proof of concept but anyone can actually continue from that.

**Comment by jillr** [comment]
We have a new way of doing these; closing and will open a new PR

**Pull request closed without merging**